### ISSUE TITLE
typing.Callable

### ISSUE THREAD

**Issue by kosyachniy** [PyCQA/pylint#5087]
### Bug description

```python
# Steps to reproduce
from typing import Callable

# Current behavior
print(isinstance([], Callable))

I get error:
`[isinstance-second-argument-not-valid-type] Second argument of isinstance is not a type`
```

### Configuration

```ini
-
```

### Command used

```shell
isinstance
```

### Pylint output

```shell
[isinstance-second-argument-not-valid-type] Second argument of isinstance is not a type
```

### Expected behavior

usual

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.2 (default, Feb 24 2021, 05:06:40) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### OS / Environment

MacOS
